### PR TITLE
feat(anta.tests): New VXLAN tests

### PR DESCRIPTION
**Pull request by carl-baillargeon**
# Description

New VXLAN tests:
- VerifyVxlanVniBinding
- VerifyVxlanVtep

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/vxlan.py [hunk 1]
```diff
@@ -73,3 +82,95 @@ def test(self) -> None:
...
+            if vni in vxlan1["vniBindings"]:
+                vlan_ = vxlan1["vniBindings"][vni]["vlan"]
+            elif vni in vxlan1["vniBindingsToVrf"]:
+                vlan_ = vxlan1["vniBindingsToVrf"][vni]["vlan"]
+            else:
+                no_binding.append(vni)
+                vlan_ = None
+
+            if vlan_ and vlan != vlan_:
+                wrong_binding.append({vni: vlan_})
```
Comment (gmuloc) [inline anta/tests/vxlan.py:128]: change `vlan_` with `retrieved_vlan` or something similar

File: anta/tests/vxlan.py [hunk 2]
```diff
@@ -73,3 +82,95 @@ def test(self) -> None:
...
+    def test(self) -> None:
+        missing_vteps = []
+
+        if (vxlan1 := get_value(self.instance_commands[0].json_output, "interfaces.Vxlan1")) is None:
+            self.result.is_skipped("Vxlan1 interface is not configured")
+            return
+
+        if len(vteps := vxlan1["vteps"]) == 0:
+            self.result.is_failure("Vxlan1 interface has no VTEP")
+            return
```
Comment (gmuloc) [inline anta/tests/vxlan.py:166]: what if `input.vteps` is an empty list ? Maybe someone wants to verify there is no VTEP peers :)

Comment (carl-baillargeon) [reply anta/tests/vxlan.py]: Fixed in 60aa0117abf83f27faed801b5710aaffb0cab147

**Review by gmuloc — APPROVED** [review]

File: tests/units/anta_tests/test_vxlan.py [hunk 3]
```diff
@@ -303,7 +303,7 @@
         "test": VerifyVxlanVtep,
         "eos_data": [{"vteps": {}, "interfaces": {"Vxlan1": {"vteps": ["10.1.1.5", "10.1.1.6"]}}}],
         "inputs": {"vteps": []},
-        "expected": {"result": "failure", "messages": ["Unexpected VTEP peer(s) on Vxlan1 interface: {'10.1.1.5', '10.1.1.6'}"]},
+        "expected": {"result": "failure", "messages": ["Unexpected VTEP peer(s) on Vxlan1 interface: {'10.1.1.6', '10.1.1.5'}"]},
```
[the remark this answers is not in the document]
Comment (carl-baillargeon) [reply tests/units/anta_tests/test_vxlan.py:306]: Ah damn I am using sets that's why CI is failing sometimes...

**Pull request merged**